src/main/java/com/ably/kafka/connect/client/DefaultAblyClient.java

```diff
@@ -26,6 +24,9 @@ public class DefaultAblyClient implements AblyClient {
     private final ChannelSinkConnectorConfig connectorConfig;
     private AblyRealtime realtime;
 
+    //When this is true, the client should abort all publishing operations and throw an exception
+    private boolean connectionFailed = false;
```

Comment:
Shouldn't this be a synchronized variable? It looks like it is being used from 2 different places/threads :thinking:

Reply:
That is a good point, I have change the type to `AtomicBoolean` for thread safety  https://github.com/ably/kafka-connect-ably/pull/65/commits/514b967879b2a93a1adc50a5d0d847f05069a099